monitor/blocksel.py

```diff
@@ -0,0 +1,105 @@
+"""allow selection of an initial block"""
+import abc
+import datetime
+
+
+class ResolveBlock(metaclass=abc.ABCMeta):
+    @classmethod
+    @abc.abstractmethod
+    def from_blockselector(cls, blocksel):
+        pass
+
+    @abc.abstractmethod
+    def resolve_block(self, w3):
+        pass
+
+
+class ResolveBlockByNumber(ResolveBlock):
+    """resolve a block by number"""
+
+    @classmethod
+    def from_blockselector(cls, blocksel):
+        return cls(int(blocksel))
+
+    def __init__(self, blocksel):
+        self.blocknr = blocksel
+
+    def resolve_block(self, w3):
+        if self.blocknr < 0:
+            blocknr = max(0, w3.eth.blockNumber + self.blocknr)
```

Comment:
```suggestion
            blocknr = max(0, w3.eth.blockNumber + self.blocknr - 1)
```

Not sure about this, but it would be weird if "-0" and "0" give vastly different results and at least in python `blocks[-1]` would refer to the last block.

Reply:
'-0' is gone.